lib/morse_translator.rb

```diff
@@ -38,6 +41,31 @@ def initialize
                     "9" => "----.",
                     "0" => "-----",
                     " " => " "}
+    @morse_code_string_array = []
+    @english_array = []
+  end
+
+  def translate(word)
+    downcase = word.downcase
+    array_of_strings = downcase.each_char.to_a
+    array_of_strings.each do |letter|
+      @morse_code_string_array << @dictionary[letter]
+    end
+    return @morse_code_string_array.join
+  end
+
+  def morse_to_eng(morse_code_string)
+    space_index = morse_code_string.index(' ')
+    index = space_index + 1
+    array_of_code = morse_code_string.split
+    array_of_code.each do |code|
+      @english_array << @dictionary.key(code)
+    end
+    final_array = @english_array.insert(index ," ")
+
+    return final_array.join
```

Comment:
Tracking he index of a double space is a nice idea here! Think that it will likely break for a phrase with more than one space. Might consider iterating through the string to find all of the double spaces and storing the index of each in an array.